website/content/platform/development/contributing/how-to/add_data_provider_extension.md

```diff
@@ -0,0 +1,269 @@
+
+An easy way to get started is to copy and paste something existing. In the OpenBB GitHub repository, provider extensions are located [here](https://github.com/OpenBB-finance/OpenBBTerminal/tree/develop/openbb_platform/providers).
+
+For convenience's sake, download this template [zip file](ttps://github.com/OpenBB-finance/OpenBBTerminal/files/14519701/provider_extension_template.zip). It contains the basic folder structure and everything required to get started creating new data models.
+
+For demonstration purposes, we'll assume this is from the perspective of somebody who will eventually open a pull request and contribute the code to the OpenBB repository. The process, however, also applies to publishing directly via PyPI. Anyone can make an OpenBB extension.
+
+The structure of the folder will look something like this:
+
+```md
+template/
+├── openbb_template/
+│   ├── models/
+│   │   ├── __init__.py
+│   │   └── some_model.py
+│   ├── utils/
+│   │   ├── __init__.py
+│   │   └── helpers.py
+│   ├── tests/
+│   │   ├── record/
+│   │   ├── __init__.py
+│   │   └── test_template_fetchers.py
+│   ├── __init__.py
+├── __init__.py
+├── pyproject.toml
+└── README.md
+```
+
+The `__init__.py` file where models are mapped to the router is under, `/openbb_template`.
+
+To get started:
+
+- Unpack the downloaded [zip](ttps://github.com/OpenBB-finance/OpenBBTerminal/files/14519701/provider_extension_template.zip) file.
+  - If working with a cloned GitHub repo, the folder is:
+  ```console
+  ~/OpenBBTerminal/openbb_platform/providers
+  ```
+- Rename everything, "template", to suit. File names, models, import statements, docstrings.
+- Add any provider-specific package requirements in the `pyproject.toml` file.
+- Update the Provider information in the `__init__.py` file.
+  - If credentials are required, add a line to the Provider class initialization.
+  ```python
+  credentials=["api_key", "account_type"], # account_type is either "sandbox" or "live"
+  ```
+- From a terminal command line, navigate into the folder where the extension is, then install the empty blank package in "editable" mode.
+  ```console
+  poetry lock
+  pip install -e .
+  ```
+- Start creating data models using the steps outlined [here](add_data_to_existing_endpoint)
+
+### Cookie Cutter
+
+In order to speed up the process of building an extension, we have created a [**Cookiecutter**](https://github.com/OpenBB-finance/openbb-cookiecutter) template.
+It serves as a jumpstart for your extension development, and can be used instead of the template ZIP referenced earlier. Instructions are located on the [GitHub page](https://github.com/OpenBB-finance/openbb-cookiecutter).
```

Comment:
Sometimes, people confuse the cookiecutter tool with developing on the platform itself. Maybe we add a note saying that you can't just copy/paste the entire content of the cookiecutter onto the platform and expect it to work without a few tweaks to the initialisation.